### PR TITLE
Feature/substitution live page

### LINKED ISSUE #92: Add online page with current substitutions.

### PR DESCRIPTION
**Pull request by KanarekLife**
fixes #92

### CODE REVIEW

File: tests/eru.Application.Tests/FakeDbContext.cs [hunk 1]
```diff
@@ -49,6 +50,31 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+                x.Property(y => y.Substituting)
+                    .HasMaxLength(255);
+                x.Property(y => y.Teacher)
+                    .HasMaxLength(255);
+            });
+
+            modelBuilder.Entity<SubstitutionsRecord>(x =>
+            {
+                x.HasKey(y => y.UploadDateTime);
+            });
```
Comment (Bazik33) [inline tests/eru.Application.Tests/FakeDbContext.cs:77]: Does not match with real 'ApplicationDbContext.cs'

File: src/eru.WebApp/Pages/Substitutions.cshtml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                        else
+                        {
+                            <td>@substitution.Substituting</td>
+                        }
+                        <td>@substitution.Note</td>
+                    </tr>
+                }
+                </tbody>
+            </table>
+    </div>
```
Comment (Bazik33) [inline src/eru.WebApp/Pages/Substitutions.cshtml:55]:
```suggestion
    <h1 class="title">@Localizer["title"]</h1>
    @if (Model.SubstitutionsRecord == null)
    {
    <h3 class="subtitle">@Localizer["content-unavailable"]</h3>
    }
    else
    {
    <h3 class="subtitle">@Localizer["uploaded-on"] @Model.SubstitutionsRecord.UploadDateTime.ToShortDateString()</h3>
    <h3 class="subtitle">@Localizer["for-date"] @Model.SubstitutionsRecord.SubstitutionsDate.ToShortDateString()</h3>
    <div class="table-container">
        <table class="table is-bordered is-striped is-fullwidth">
                <thead>
                <tr>
                    <th>@Localizer["lesson"]</th>
                    <th>@Localizer["classes"]</th>
                    <th>@Localizer["subject"]</th>
                    <th>@Localizer["teacher"]</th>
                    <th>@Localizer["groups"]</th>
                    <th>@Localizer["room"]</th>
                    <th>@Localizer["details"]</th>
                    <th>@Localizer["note"]</th>
                </tr>
                </thead>
                <tbody>
                @foreach (var substitution in Model.SubstitutionsRecord.Substitutions)
                {
                    <tr>
                        <td>@substitution.Lesson</td>
                        <td>@string.Join(';',substitution.Classes)</td>
                        <td>@substitution.Subject</td>
                        <td>@substitution.Teacher</td>
                        <td>@substitution.Groups</td>
                        <td>@substitution.Room</td>
                        @if (substitution.Cancelled)
                        {
                            <td>@Localizer["canceled"]</td>
                        }
                        else
                        {
                            <td>@substitution.Substituting</td>
                        }
                        <td>@substitution.Note</td>
                    </tr>
                }
                </tbody>
            </table>
    </div>
    }
```

When there are no substitutions in database application throws an [exception](https://imgur.com/a/7Dd5w2Y).

**Review by Bazik33 — CHANGES_REQUESTED** [review]
(1) There is a bug in UploadSubstitutionsCommandHandler: a class that IS NOT in DbContext BUT has been just added (so it's in newClasses List) will never be added to trackedClasses. Thus, lines from 32 to 52 should look like this: 
```csharp
foreach (var @class in classes)
{
    var tmpClass = await _context.Classes.FirstOrDefaultAsync(x => x.Year == @class.Year && x.Section == @class.Section, cancellationToken);
    if (tmpClass == null && !newClasses.Any(x=>x.Year == @class.Year && x.Section == @class.Section))
    {
        var entity = await _context.Classes.AddAsync(@class, cancellationToken);
        newClasses.Add(entity.Entity);
        trackedClasses.Add(entity.Entity);   
    }
    else
    {
        trackedClasses.Add(tmpClass);
    }
}
```

(2) You can consider deleteing some migrations you've created. This will help keep the project clean.

**Comment by KanarekLife** [comment]
Without Id on Substitution and auto generation of Id in FakeDbContext tests couldn't be runned so I had to fix it.

**Comment by KanarekLife** [comment]
Okey I managed to fix it ;)

File: src/eru.WebApp/appsettings.json [hunk 3]
```diff
@@ -11,9 +11,8 @@
   },
   "AllowedHosts": "*",
   "Database": {
-    "Type": "sqlite",
     "AutomaticallyMigrate": true,
-    "ConnectionString": "Data Source=eru.db"
+    "ConnectionString": "Host=localhost;Username=eru;Password=Minecraft5066!;Database=eru"
```
Comment (Bazik33) [inline src/eru.WebApp/appsettings.json:15]:
```suggestion
    "ConnectionString": "Host=myserver;Username=mylogin;Password=mypass;Database=mydatabase;"
```

**Review by Bazik33 — APPROVED** [review]

**Pull request merged**